### ISSUE TITLE
viewsets.ModelViewSet causes timeout

### ISSUE THREAD

**Issue by variable** [tomchristie/django-rest-framework#3629]
The following code causes the server to hang by visiting GET <code>/gopermissions/</code>
``` python
class GoPermissionViewSet(viewsets.ModelViewSet):
    permission_classes = [HasGroupPermission]
    required_groups = {
        'GET': ['grabone'],
        'POST': ['grabone'],
    }
    serializer_class = GoPermissionSerializer
    queryset = GoPermission.objects.all()
    model = GoPermission
    lookup_field = 'uuid'

```
We have over 1 million users, and when I run show processlist on mysql, this comes up:
<code>
SELECT `auth_user`.`id`, `auth_user`.`password`, `auth_user`.`last_login`, `auth_user`.`is_superuser
</code>

But once I have changed to the following
```python
class GoPermissionViewSet(mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet):
```

Works without problem.

**Comment by xordoquy** [comment]
You likely have an issue with your database.

Without a detailed analysis relating to where / how Django REST framework does something wrong there isn't much we can do.

**Comment by variable** [comment]
this is the model:
```python
class GoPermission(TimeStampedMixin, models.Model):
    user = models.ForeignKey(User)
    namespace = models.SlugField()
    permission = models.SlugField()
    object_type = models.SlugField()
    object_id = models.SlugField()
```
if I exclude the <code>user</code> in the serializer, the ModelViewSet will work
```python
class GoPermissionSerializer(serializers.ModelSerializer):
    class Meta:
        model = GoPermission
        exclude = ['user']
```

**Comment by xordoquy** [comment]
@variable and you still provide no proof DRF is doing something wrong. This sounds to me like a non optimized database. Unless you can identify the performance issue in DRF I don't see a reason to reopen.

**Comment by tomchristie** [comment]
Let me rephrase this. The issue here is almost certainly some aspect of your app configuration that's causing all the user records to be scanned. Perhaps that's due to serialising them all, perhaps a case where you need a select_related, or something along those lines. What this doesn't particularly appear to be tho (given the current amount of info at least) is a bug in REST framework.

Sympathetic that it's frustrating & we can try to help with configuration issues, but you'll need to try to reduce the case to something simpler first so it's more obvious where the issue lies.

Cut the problem down to something more isolated & obviously reproducible and we may be able to help more.

**Comment by variable** [comment]
I have mentioned here https://github.com/tomchristie/django-rest-framework/issues/3630, I think it's quite obvious that CreateModelMixin is causing the problem, but why does CreateModelMixin stop a simple GET request is unknown until I dig into DRF's source code.

If you take a look at info I mentioned in above link, all I had to do for a work around was to create a User serializer with no customization:

```python
class UserSerializer(serializers.ModelSerializer):

    class Meta:
        model = User
```

By the sound of it I might just have to read into the source code myself.

**Comment by variable** [comment]
Maybe I have an answer to my own problem.  I think it's caused by the APIView that tries to create a POST form, if I don't define a serializer for user, it will try to render a drop down list with 1+ million plus users as options.

**Comment by xordoquy** [comment]
@variable this can be verified by requesting a JSON response. You can do that by setting the content type to application/json. If the request is fast enough then it's likely linked to the browsable API rendering the User in which case you likely want to use a [different method](http://www.django-rest-framework.org/topics/browsable-api/#autocomplete) to render the User combo box

**Comment by xordoquy** [comment]
@variable I'm sorry I was rude to you. I should have mentioned that we try to keep usage question out of the issue tracker. Those are better asked on the [mailing list](https://groups.google.com/forum/?fromgroups#!forum/django-rest-framework) or stack overflow.

**Comment by tomchristie** [comment]
@variable - We do a better job of guarding against this in recent versions and will only render a maximum of 1000 items in a select HTML control for a relationship. See here: http://www.django-rest-framework.org/api-guide/relations/#select-field-cutoffs

**Comment by variable** [comment]
All good, thanks guys for the heads up, I am wondering if the select-field-cutoffs works if no custom serializer is fined, I think it wouldn't have caused timeout problem if it's only 1000 records.
